engine/src/main/java/org/hibernate/search/engine/mapper/schema/management/SchemaExport.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.mapper.schema.management;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+
+import org.hibernate.search.engine.mapper.schema.management.impl.FileSchemaElementExporter;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+@Incubating
+public interface SchemaExport {
+	/**
+	 * Applies the provided exporter to each schema element context that is represented by this schema export.
+	 *
+	 * @param elementExporter The exporter to apply to each element.
+	 */
+	void applyElementExporter(Consumer<SchemaElementContext> elementExporter);
+
+	/**
+	 * A simple exporter that writes the schema out to the provided target directory. Renders a file tree containing
+	 * all backends that this export represents and their corresponding indexes.
+	 *
+	 * @param targetDirectory The target directory to generate the output into.
+	 * @return The exporter that should be provided to {@link SchemaExport#applyElementExporter(Consumer)}
+	 */
+	static Consumer<SchemaElementContext> fileExporter(Path targetDirectory) {
+		return new FileSchemaElementExporter( targetDirectory, SchemaElementContext.OutputFormat.SIMPLE );
+	}
```

Comment:
Or simply expose a `toFiles` method?

```suggestion
	/**
	 * Writes the schema out to the provided target directory. Renders a file tree containing the schema of
	 * all backends that this export represents and their corresponding indexes.
	 *
	 * @param targetDirectory The target directory to generate the output into.
	 * @return The exporter that should be provided to {@link SchemaExport#applyElementExporter(Consumer)}
	 */
	default void toFile(Path targetDirectory) {
		applyElementExporter( new FileSchemaElementExporter( targetDirectory, SchemaElementContext.OutputFormat.SIMPLE ) );
	}
```